src/Categories/Category/Instance/Simplex.agda

```diff
@@ -1,67 +1,152 @@
 {-# OPTIONS --without-K --safe #-}
 
-open import Categories.Category
 
+-- The Simplex category Δ
 module Categories.Category.Instance.Simplex where
 
-open import Level
+open import Level using (0ℓ)
 open import Data.Product
-open import Data.Fin.Base using (Fin; _≤_)
-open import Data.Nat.Base using (ℕ; z≤n; s≤s)
+open import Data.Fin.Base using (Fin; zero; suc; _≤_; _<_; inject₁)
+open import Data.Nat.Base using (ℕ; zero; suc; z≤n; s≤s)
 open import Function renaming (id to idF; _∘_ to _∙_)
 
-open import Relation.Binary using (_=[_]⇒_)
-open import Relation.Binary.PropositionalEquality
+open import Relation.Binary.PropositionalEquality using (_≡_; refl; sym; trans; cong)
+
+open import Categories.Category.Core using (Category)
+
+--------------------------------------------------------------------------------
+-- The Simplex Category
+--
+-- Classically, the category Δ is defined as follows:
+--   * Objects are Natural Numbers
+--   * Morphisms 'm ⇒ n' are order-preserving maps 'Fin m → Fin n'
+--
+-- We _can_ define this version of Δ, but it is generally /not/ the version we want.
+-- Most of the time we work with Δ, we only care about 2 classes of maps:
+--   * Face maps 'δᵢ : n ⇒ (1 + n)' whose image omits 'i'
+--   * Boundary maps 'σᵢ : (1 + n) ⇒ n' such that 'σᵢ i = σᵢ (i + 1) = i'
+-- It turns out that all of the maps of Δ can be decomposed into compositions
+-- of these 2 classes of maps[1], so most texts will define Functors out of Δ
+-- based solely how they act on these morphisms.
+--
+-- Therefore, we want our definition of Δ to put these morphisms front and center,
+-- by encoding our homs as follows:
+
+infix 4 _Δ⇒_
+infixr 9 _⊚_
+
+data _Δ⇒_ : ℕ → ℕ → Set where
+  ε  : ∀ {n} → n Δ⇒ n
+  δ  : ∀ {n} → (i : Fin (suc n)) → n Δ⇒ (suc n)
+  σ  : ∀ {n} → (j : Fin n) → (suc n) Δ⇒ n
+  _⊚_ : ∀ {l m n} → m Δ⇒ n → l Δ⇒ m → l Δ⇒ n
+
+-- However, this raises some tricky questions about equality of morphisms. It is tempting
+-- to use the Simplical Identities[2]:
+--   * δᵢ ∘ δⱼ = δⱼ₊₁ ∘ δᵢ if i ≤ j
+--   * σⱼ ∘ σᵢ = σᵢ ∘ σⱼ₊₁ if i ≤ j
+--   * σⱼ ∘ δᵢ = δᵢ ∘ σⱼ₋₁ if i < j
+--   * σⱼ ∘ δᵢ = id        if i = j or i = j + 1
+--   * σⱼ ∘ δᵢ = δᵢ₋₁ ∘ σⱼ if j + 1 < i
+--
+-- However, we can do better than this. Instead, we appeal to the denotational
+-- semantics of our formal chains of morphisms. In this case we interpret
+-- our morphisms as maps between finite ordinals:
+
+face : ∀ {n} → Fin (ℕ.suc n) → Fin n → Fin (ℕ.suc n)
+face Fin.zero    k           = Fin.suc k
+face (Fin.suc i) Fin.zero    = Fin.zero
+face (Fin.suc i) (Fin.suc k) = Fin.suc (face i k)
```

Comment:
Good to know! Perhaps it makes sense to add `degen` to the stdlib as well under a less domain-specific name.

Reply:
I agree - if you can figure out a good name. I couldn't...

Reply:
https://github.com/agda/agda-stdlib/pull/1492

Reply:
I understand that it will make sense to still have the names `face` and `degen` in this file, right? So there is no need to wait for the standard library to merge these in. The definition of `face` can be shortened now, and that of `degen` later.

Reply:
Yeah, I was still going to give them new names here for clarity, so we aren't blocked on that PR.